packages/amplify-graphql-transformer/src/graphql-transformer.ts

```diff
@@ -176,8 +176,15 @@ export const executeTransform = (config: ExecuteTransformConfig): void => {
     assetProvider,
     synthParameters,
     parameterProvider,
+    transformersFactoryArgs,
   } = config;
 
+  const mergedSynthParameters = {
```

Comment:
Since I don't fully understand the construct arch I'm confused why we define the roles & IDs as both synth params and this factory args override?